src/Microsoft.OData.Core/ODataMessageWriter.cs

```diff
@@ -1212,6 +1212,13 @@ private void Dispose(bool disposing)
                     {
                         this.outputContext.Dispose();
                     }
+                    else if (!this.writeMethodCalled)

```

Comment:
is the !this.writeMethodCalled check needed? Is there a scenario where outputContext == null and this.writeMethodCalled == true?